snapshots/input/syntax/src/structural-type.ts

```diff
@@ -0,0 +1,9 @@
+export function foo(): Promise<{ member: number }> {
+  return Promise.resolve({ member: 42 })
+}
```

Comment:
What happens when there is a mapped type (https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) involved as part of the definition? A mapped type can produce new object types, so the keys aren't visible in the source code, they're implicit as part of some logic.

E.g. if you have:

```
type OptionsFlags<Type> = {
  [Property in keyof Type]: boolean;
};

type FeatureFlags = {
  darkMode: () => void;
};
 
type FeatureOptions = OptionsFlags<FeatureFlags>;
// implicitly 
// type FeatureOptions = {
//    darkMode: boolean;
// }

const fo: FeatureOptions = { darkMode: true };
                          // ^ go to def
```

Will go-to-def on `darkMode` in the object literal take you to `FeatureOptions`? Will it take you to `FeatureFlags`? Nowhere?

Reply:
Nice one. It doesn't work. I added a snapshot test and opened an issue to follow up on this https://github.com/sourcegraph/scip-typescript/issues/187